hasher-matcher-actioner/webapp/src/pages/MatchIndex.jsx

```diff
@@ -0,0 +1,251 @@
+  const searchQuery = query.get('contentId') || query.get('signalId');
+  let searchInAttribute;
+
+  ['contentId', 'signalId'].forEach(attribute => {
+    if (query.has(attribute)) {
+      searchInAttribute = attribute;
+    }
+  });
+
+  const [
+    selectedContentAndSignalIds,
+    setSelectedContentAndSignalIds,
+  ] = useState({
+    contentId: undefined,
+    signalId: undefined,
+    signalSource: undefined,
+  });
+
+  return (
+    <div className="d-flex flex-column justify-content-start align-items-stretch h-100 w-100">
+      <div className="flex-grow-0">
+        <Container className="bg-dark text-light" fluid>
+          <Row className="d-flex flex-row justify-content-between align-items-end">
+            <div className="px-4 py-2">
+              <h1>Matches</h1>
+              <p>View content that has been flagged by the HMA system.</p>
+            </div>
+            <div className="px-4 py-2">
+              <MatchListFilters
+                searchInAttribute={searchInAttribute}
+                searchQuery={searchQuery}
+              />
+            </div>
+          </Row>
+        </Container>
+      </div>
+      <div className="flex-grow-1" style={{overflowY: 'hidden'}}>
+        <Container className="h-100 v-100" fluid>
+          <Row className="d-flex align-items-stretch h-100">
+            <Col className="h-100" style={{overflowY: 'auto'}} md={6}>
+              <MatchList
+                selection={selectedContentAndSignalIds}
+                onSelect={setSelectedContentAndSignalIds}
+                searchInAttribute={searchInAttribute}
+                searchQuery={searchQuery}
+              />
+            </Col>
+            <Col md={6}>
+              {(selectedContentAndSignalIds.contentId === undefined && (
+                <p>Select a match to see its details.</p>
+              )) || (
+                <ContentMatchPane
+                  contentId={selectedContentAndSignalIds.contentId}
+                  signalId={selectedContentAndSignalIds.signalId}
+                  signalSource={selectedContentAndSignalIds.signalSource}
+                />
+              )}
+            </Col>
+          </Row>
+        </Container>
+      </div>
+    </div>
+  );
+}
+
+/**
+ * A box of filters for the list of matches that appear on the match filters page.
+ */
+function MatchListFilters({searchInAttribute, searchQuery}) {
+  const [localSearchInAttribute, setLocalSearchInAttribute] = useState(
+    searchInAttribute,
+  );
+  const [localSearchQuery, setLocalSearchQuery] = useState(searchQuery);
+
+  useEffect(() => {
+    /** Update state when props changed. */
+    setLocalSearchInAttribute(searchInAttribute);
+    setLocalSearchQuery(searchQuery);
+  }, [searchInAttribute, searchQuery]);
+
+  const history = useHistory();
+  const loadSearchPage = e => {
+    history.push(`?${localSearchInAttribute}=${localSearchQuery}`);
+    e.preventDefault(); // Prevent form submit
+  };
+
+  return (
+    <Form>
```

Comment:
nit: I think it would be good to have a more intuitive 'clear' filter method then having to delete the text.